### ISSUE TITLE
compiletest: technically potential false positives for filecheck annotations

### ISSUE THREAD

**Issue by jieyouxu** [rust-lang/rust#125825]
This is a fun one: in https://github.com/rust-lang/rust/pull/121561 I added checks to error if we detect any legacy-directives-like comments (which are of the form `// <known_directive_name>`.

In codegen tests, we have filecheck directives that can be revisioned:

```rs
//@ revisions: known_directive_name
// known_directive_name: noundef
```

... but unfortunately, `// known_directive_name` looks like a legacy directive to compiletest so we might error here even though it is not actually a legacy directive (it just "looks" like one for the heuristic I added).

Technically a bug (false positive), but in practice changing the revision name to something else would sidestep this check.

Note: the legacy directive check would also fire on

```
// this is a block of comment that has a
// ignore in the middle of the sentence
// that looks like a legacy directive
```

**Comment by jieyouxu** [comment]
Realized this in https://github.com/rust-lang/rust/issues/125813#issuecomment-2142539708, the long term solution is probably to have compiletest directive handling be cooperative between "base" directives and test suite "customized" directives.

**Comment by workingjubilee** [comment]
fwiw despite noting the issue, I think the prefix doesn't matter too much as FileCheck is very simple and I don't think it is aware of any comment styles itself. I think it literally just scans for the prefix on a line and assumes you won't introduce code that resembles a FileCheck directive.

**Comment by jieyouxu** [comment]
compiletest triage: I deleted the legacy directive check so this is no longer a problem.
